### PR TITLE
Assignment 4 Isabel Escribano

### PR DESCRIPTION
**Pull request by IsaEscman**

### CODE REVIEW

File: Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+ns = Namespace("http://somewhere#")
+from rdflib.plugins.sparql import prepareQuery
+
+livingThing_query = prepareQuery('''
+SELECT ?subClase
+WHERE {
+    ?subClase rdfs:subClassOf ns:LivingThing
```
Comment (SerPablo) [inline Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py:33]: subClassOf+

File: Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+''',
+initNs = { "rdfs": RDFS, "ns": ns}
+)
+
+# Visualize the results
+for r in g.query(livingThing_query):
+  print(r.subClase)
+
```
Comment (SerPablo) [inline Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py:42]: Add a RDFlib solution

File: Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+WHERE {
+    ?person rdf:type ns:Person
+}
+''',
+initNs = { "rdf": RDF, "rdfs": RDFS, "ns": ns}
+)
+# Visualize the results
+for r in g.query(person_query):
+  print(r.person)
+
```
Comment (SerPablo) [inline Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py:60]: Add a rdflib solution

File: Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+ns = Namespace("http://somewhere#")
+person_query = prepareQuery('''
+SELECT ?person
+WHERE {
+    ?person rdf:type ns:Person
```
Comment (SerPablo) [inline Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py:52]: check the subclasses
Hint a/rdfs:subClassOf*

File: Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+     ?type = ns:Animal || ?type = ns:Person
+)
+}
+''',
+initNs = { "rdf": RDF, "rdfs": RDFS, "ns": ns}
+)
+# Visualize the results
+for r in g.query(query_3):
+  print(r.subject, r.predicate, r.object)
+
```
Comment (SerPablo) [inline Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py:82]: Check the result of this query to not get duplicates. The result should be:
[http://somewhere#SaraJones](http://somewhere/#SaraJones)
[http://somewhere#JohnSmith](http://somewhere/#JohnSmith)
[http://somewhere#RockySmith](http://somewhere/#RockySmith)

File: Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+query_4 = prepareQuery('''
+SELECT ?Rocky_URI  ?amigo_URI ?amigo_name
+WHERE {
+     ?Rocky_URI <http://www.w3.org/2001/vcard-rdf/3.0/Given>  ?given_name .
+     ?amigo_URI foaf:knows ?Rocky_URI .
+     ?amigo_URI <http://www.w3.org/2001/vcard-rdf/3.0/Given>  ?amigo_name .
+
+
+     FILTER(?given_name = "Rocky")
+}
```
Comment (SerPablo) [inline Assignment4/Isabel Escribano - e22c048/task07_Isabel_EM.py:99]: Im not sure about this result, It prints Sara and Jane and should be only Sara

**Review by SerPablo — COMMENTED** [review]
Check the comments. Task7 should provide the results in rdflib and sparql

**Comment by anaigmo** [comment]
Entregado sin corregir, no se admiten más PR

**Pull request merged**